vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt

```diff
@@ -67,11 +78,40 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
 
             // Schedule a timer to stop sharing
             scheduleTimer(roomArgs.roomId, roomArgs.durationMillis)
+
+            // Send beacon info state event
+            activeSessionHolder
+                    .getSafeActiveSession()
+                    ?.let { session ->
+                        session.coroutineScope.launch {
```

Comment:
Maybe it would be better to precise the IO Dispatcher here like this:
```
session.coroutineScope.launch(session.coroutineDispatchers.io) {
    sendBeaconInfo(session, roomArgs)
}
```
I am not sure but reading the implementation of [SessionCoroutineScopes](https://github.com/vector-im/element-android/blob/98177da6b1e1e2d00728ec626ce3ce767efdeec6/vector/src/main/java/im/vector/app/features/session/SessionCoroutineScopes.kt), it seems the default Dispatcher is the Main one. @ouchadam or @bmarty could you confirm? Or am I missing something?

Reply:
`Session.getRoom()` is a blocking non suspend call, ideally the api would be suspending and internally use the correct dispatcher where needed, until then wrapping with an IO context would be better! 

I thought this would have triggered a network on main thread warning :thinking: